src/vs/workbench/api/common/extHost.api.impl.ts

```diff
@@ -755,7 +756,10 @@ export function createApiFactoryAndRegisterActors(accessor: ServicesAccessor): I
 				return extHostDocumentContentProviders.registerTextDocumentContentProvider(scheme, provider);
 			},
 			registerTaskProvider: (type: string, provider: vscode.TaskProvider) => {
-				throw new Error('Tasks api is not allowed in Azure Data Studio'); // {{SQL CARBON EDIT}} disable task
+				extHostApiDeprecation.report('window.registerTaskProvider', extension,
```

Comment:
Just for my understanding if we never had this what are we starting this with a deprecated message - shouldn't we just directly use the new non-deprecated flow?

Reply:
This is from VS Code